File: second-year/second-semester/Ingineria Sistemelor Soft/MonitoringSystem/monitoringSystem/src/main/java/com/monitoringsystem/service/api/UserService.java
package com.monitoringsystem.service.api;

import com.monitoringsystem.model.User;
import com.monitoringsystem.utils.Observable;

import java.util.List;
import java.util.Optional;

public interface UserService extends Observable {
    void add(String username, String password, String firstName, String lastName, Integer role);
    void delete(Long id);
    Optional<User> findUser(String username, String password);
    List<User> getAll();

    void logout();
}

File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/travelAgency2/Service/UserService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using travelAgency2.Repository;
using travelAgency2.src.Domain;

namespace travelAgency2.Service
{
    internal class UserService
    {
        private UserRepository userRepository;
        private Random random;

        public UserService(UserRepository userRepository)
        {
            this.userRepository = userRepository;
            this.random = new Random();
        }

        void Add(string username, string password, string firstName, string lastName)
        {
            long id = random.Next(0, 999999999);
            User user = new User(id, username, password, firstName, lastName);
            userRepository.Add(user);
        }

        User FindById(long id)
        {
            return userRepository.FindByID(id);
        }

        List<User> GetAll()
        {
            return userRepository.GetAll();
        }

        public User FindUser(string username, string password)
        {
            return userRepository.findUser(username, password);
        }
    }
}
